src/com/owncloud/android/ui/adapter/LocalFileListAdapter.java

```diff
@@ -104,6 +107,12 @@ public View getView(int position, View convertView, ViewGroup parent) {
             fileName.setText(name);
             
             ImageView fileIcon = (ImageView) view.findViewById(R.id.thumbnail);
+
+            /** Cancellation needs do be checked and done before changing the drawable in fileIcon, or
+             * {@link ThumbnailsCacheManager#cancelPotentialWork} will NEVER cancel any task.
+             **/
+            boolean allowedToCreateNewThumbnail = (ThumbnailsCacheManager.cancelPotentialWork(file, fileIcon));
```

Comment:
What is the reason for having this flag in this line? Wouldn't this flag only be needed in the else branch (!directory), so for a directory you wouldn't initialize and calculate the flag, just in the branch for files? :)